src/borg/archiver.py

```diff
@@ -973,45 +992,71 @@ def format_cmdline(cmdline):
             if not archive_names:
                 return self.exit_code
 
+        output_data = []
+
         for i, archive_name in enumerate(archive_names, 1):
             archive = Archive(repository, key, manifest, archive_name, cache=cache,
                               consider_part_files=args.consider_part_files)
-            stats = archive.calc_stats(cache)
-            print('Archive name: %s' % archive.name)
-            print('Archive fingerprint: %s' % archive.fpr)
-            print('Comment: %s' % archive.metadata.get('comment', ''))
-            print('Hostname: %s' % archive.metadata.hostname)
-            print('Username: %s' % archive.metadata.username)
-            print('Time (start): %s' % format_time(to_localtime(archive.ts)))
-            print('Time (end):   %s' % format_time(to_localtime(archive.ts_end)))
-            print('Duration: %s' % archive.duration_from_meta)
-            print('Number of files: %d' % stats.nfiles)
-            print('Command line: %s' % format_cmdline(archive.metadata.cmdline))
-            print('Utilization of max. archive size: %d%%' % (100 * cache.chunks[archive.id].csize / MAX_DATA_SIZE))
-            print(DASHES)
-            print(STATS_HEADER)
-            print(str(stats))
-            print(str(cache))
+            if args.json:
+                output_data.append(archive.info())
+            else:
+                stats = archive.calc_stats(cache)
+                print('Archive name: %s' % archive.name)
+                print('Archive fingerprint: %s' % archive.fpr)
```

Comment:
This is a bit more difficult, because human formatting differs from JSON formatting. So if one uses the dict used for JSON, then one has to reformat keys after the fact or not really use the dict for everything, so I didn't.